### PR TITLE
fixed assorted ore bag/box bugs and perf issues

### PR DESCRIPTION
**Pull request by Spookerton**
mining satchels automatically collect ore if held in the hands, belt, or suit slot.

pulling an ore box with a satchel collects ore directly into the ore box instead of the satchel, and empties the satchel if appropriate.

hostile mining drones collect into ore boxes just like people would. Their collection checking view size is smaller.

mining satchels and ore boxes only update their contents hints if examined after it changes, rather than up to once a second regardless of changes.

various doubled messages removed.

removed various dm-land loops in favor of contents addition where it makes sense.

### CODE REVIEW

File: code/modules/mining/ore_box.dm [hunk 1]
```diff
@@ -1,96 +1,102 @@
...
-	if(!contents.len)
-		. += "It is empty."
-		return .
 
-	if(world.time > last_update + 10)
-		update_ore_count()
-		last_update = world.time
+/obj/structure/ore_box/ex_act(severity)
+	var/turf/turf = get_turf(src)
+	switch (severity)
```
[the remark this answers is not in the document]
Comment (Spookerton) [reply code/modules/mining/ore_box.dm:67]: Then it is ignored.

File: code/game/objects/items/weapons/storage/bags/ore.dm [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+		item.forceMove(get_turf(src))
+	item.on_exit_storage(src)
+	stored_ore_dirty = TRUE
+	update_icon()
+	return TRUE
+
+
+/obj/item/storage/bag/ore/gather_all(turf/from, mob/living/user, silent, autoload)
+	var/obj/structure/ore_box/box = user.pulling
+	if (istype(box))
```
[the remark this answers is not in the document]
Comment (Spookerton) [reply code/game/objects/items/weapons/storage/bags/ore.dm:76]: Having a global scope proc to generalize this is possible, but fugly in my opinion.

File: code/game/objects/items/weapons/storage/bags/ore.dm [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+		stored_ore = null
+		var/list/ores = list()
+		for (var/obj/item/ore/ore in contents)
+			++ores[ore.name]
+		var/list/chunks = list()
+		for (var/name in ores)
+			chunks += "[ores[name]] [name]"
+		if (length(chunks))
+			var/full = length(contents) >= max_storage_space
+			stored_ore = "[full ? "It is <b>full</b>! " : ""]It contains [english_list(chunks)]"
```
[the remark this answers is not in the document]
Comment (Spookerton) [reply code/game/objects/items/weapons/storage/bags/ore.dm:36]: whoops!

[the remark this answers is not in the document]
Comment (Spookerton) [reply code/game/objects/items/weapons/storage/bags/ore.dm:36]: Sure you can. The english_list part is always included.

File: code/modules/mining/ore_box.dm [hunk 4]
```diff
@@ -1,96 +1,102 @@
...
+		if (1)
+			if (turf == loc)
+				turf.contents += contents
```
Comment (Spookerton) [inline code/modules/mining/ore_box.dm:70]: The reason for this pattern is simple performance. A forceMove or dropInto loop would be (and is) horribly painful by comparison.

File: code/game/objects/items/weapons/storage/bags/ore.dm (hunk 3 above)
[the remark this answers is not in the document]
Comment (Atermonera) [reply code/game/objects/items/weapons/storage/bags/ore.dm:36]: Ah, right, just not reading the braces around hte ternary correctly.

**Review by Atermonera — APPROVED** [review]

**Pull request merged**